hugo/assets/ts/widgets/search-filter.ts

```diff
@@ -139,21 +140,40 @@ export class SearchFilter extends BaseWidget {
         this.updateUrl();
     }
 
-    private updateUrl() {
-        window.location.href = `${ window.location.href.split('?')[0] }?${ queryToUrlParams(this.parsedQuery) }`;
+    private updateUrl(refresh = true) {
+        const url = `${ window.location.href.split('?')[0] }?${ queryToUrlParams(this.parsedQuery) }`;
+        if (!refresh) {
+            window.history.pushState(null, '', url);
+        } else {
+            window.location.href = url;
+        }
     }
 
-    private getParamsFromUrl(): void {
+    private getParams(): void {
+        // Get url params
         const url = new URL(window.location.href);
         const searchParams = new URLSearchParams(url.search);
         const query = searchParams.get('q');
 
         this.parsedQuery = parseQuery(query);
 
-        if (!query) {
-            this.selectedItems = [];
-        } else {
-            this.selectedItems = this.parsedQuery.facets[this.filterName] ?? [];
+        let urlParamsChanged = false;
+        let selectedItems = this.parsedQuery.facets[this.filterName] ?? [];
+
+        // Get preselected items if no items set in url (if set in urL: url wins from preselected)
+        if (selectedItems.length === 0 && this.element.dataset.preselectedItems && this.element.dataset.preselectedItems !== '') {
```

Comment:
As discussed in our call: we should only add preselected tags  when there is no query at all and not just when there is no tags in the query. So change it to if !(query) I guess or !this.parsedQuery ( not sure what the value is when there is no query)

Reply:
Should be updated now.